src/windows.c

```diff
@@ -190,6 +195,14 @@ struct pthreadpool* pthreadpool_create(size_t threads_count) {
 	return threadpool;
 }
 
+void pthreadpool_set_max_num_threads(struct pthreadpool* threadpool, size_t num_threads) {
+	pthread_mutex_lock(&threadpool->execution_mutex);
```

Comment:
`threadpool->execution_mutex` is a WinAPI mutex handle, use `WaitForSingleObject`/`ReleaseMutex`

Reply:
Aah. Thanks for pointing this out. Surprised that internal windows build did not fail. I will make sure to build this on windows.